Assets/MixedRealityToolkit.Extensions/Webrtc/Scripts/Webrtc.cs

```diff
@@ -0,0 +1,369 @@
+        {
+            /// <summary>
+            /// Indicates there is no Ice information
+            /// </summary>
+            /// <remarks>
+            /// Under normal use, this should not be used
+            /// </remarks>
+            None = 0,
+
+            /// <summary>
+            /// Indicates Ice information is of type STUN
+            /// </summary>
+            /// <remarks>
+            /// https://en.wikipedia.org/wiki/STUN
+            /// </remarks>
+            Stun,
+
+            /// <summary>
+            /// Indicates Ice information is of type TURN
+            /// </summary>
+            /// <remarks>
+            /// https://en.wikipedia.org/wiki/Traversal_Using_Relays_around_NAT
+            /// </remarks>
+            Turn
+        }
+
+        /// <summary>
+        /// Represents an Ice server in a simple way that allows configuration from the unity inspector
+        /// </summary>
+        [Serializable]
+        public struct ConfigurableIceServer
+        {
+            /// <summary>
+            /// The type of the server
+            /// </summary>
+            public IceType Type;
+
+            /// <summary>
+            /// The unqualified uri of the server
+            /// </summary>
+            /// <remarks>
+            /// You should not prefix this with "stun:" or "turn:"
+            /// </remarks>
+            public string Uri;
+
+            /// <summary>
+            /// Convert the server to the representation the underlying libraries use
+            /// </summary>
+            /// <returns>stringified server information</returns>
+            public override string ToString()
+            {
+                return string.Format("{0}: {1}", Type.ToString().ToLower(), Uri);
+            }
+        }
+
+        /// <summary>
+        /// A UnityEvent that represents error events
+        /// </summary>
+        [Serializable]
+        public class WebrtcErrorEvent : UnityEvent<string>
+        {
+        }
+
+        /// <summary>
+        /// Retrieves the underlying peer
+        /// </summary>
+        /// <remarks>
+        /// If <see cref="OnInitialized"/> has not fired, this will be <c>null</c>
+        /// </remarks>
+        public IPeer Peer
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// Flag to initialize Webrtc on <see cref="Start"/>
+        /// </summary>
+        [Tooltip("This will initialize Webrtc on Start()")]
+        public bool AutoInitialize = true;
+
+        /// <summary>
+        /// Flag to log all errors to the Unity console automatically
+        /// </summary>
+        [Tooltip("This will log all errors to the Unity console")]
+        public bool AutoLogErrors = true;
+
+        /// <summary>
+        /// Set of ice servers
+        /// </summary>
+        [Tooltip("(Optional) Set of Ice servers")]
+        public List<ConfigurableIceServer> IceServers = new List<ConfigurableIceServer>()
```

Comment:
We should support Ice-credentials along with the server list coming from the signaling service as a part of the negotiation/authentication.

Reply:
this is unfortunately not trivial (as the native library also takes information in this odd format at this time). I will book a feature request for it but i will not be able to refactor to this myself at this point in time.

Reply:
captured (and tagged) in comment below, as it is unclear at this time if GH issues here is the right place to capture this work.

Reply:
The format is fine. All we need is a way to accept [TURN Auth message](https://www.ietf.org/proceedings/87/slides/slides-87-behave-10.pdf)

```json
{
  username: "1375043478:abcd1234",
  password: <HMAC("1375043487:abcd1234", SS)>
  ttl: 86400,
  uris: [
    "turn:turn.bar.com:3478?proto=udp",
    "turn:turn.bar.com:3478?proto=tcp",
    "turns:turn.bar.com:443?proto=tcp"
] }
```
I can make necessary changes. Although, it’s important to keep existing properties too, so developer can provide values event if signaling server gives nothing.

Reply:
> TLDR: sure, that sounds great. I'll look to you for the changset to allow a signaling server to provide this info.

ah - thought you were proposing supporting giving native library something like:

```
NativePeer(List<IceServer> servers)
```

where

```
struct IceServer
{
    string Uri;
    int Port;
    string Username;
    string Credential;
}
```

which imo is a more common way to describe a set of ice servers. Current native lib only supports once `Username` and `Credential` even if N ice servers urls is given.